src/js/flows/submitArchiveSearch.js

```diff
@@ -13,8 +13,9 @@ import whenIdle from "../lib/whenIdle"
 export default (patterns: string[]): Thunk => (
   dispatch,
   getState,
-  {zealot}
+  {createZealot}
 ) => {
+  const zealot = createZealot(Current.getConnectionId(getState()))
```

Comment:
I wonder if it would be worth dispatching a selector to make this more conscise, since we're doing it so many places.

```js
const getZealot = () => (dispatch, getState, {createZealot}) => {
  return createZealot(Current.getConnectionId(getState()))
}
```

```js
// then in all the places we could just do
const zealot = dispatch(getZealot())
```